File: pratice/LogicalCoding/MathProblems.js
function fibonacciSeries(n) {
  let prev1 = 0,
    prev2 = 1,
    current;
  console.log(prev1);
  console.log(prev2);
  for (let i = 2; i <= n; i++) {
    current = prev1 + prev2;
    console.log(current);
    prev1 = prev2;
    prev2 = current;
  }
}

function factorial(n) {
  let fact = 1;

  for (let i = 2; i <= n; i++) {
    fact *= i;
  }
  return console.log(fact);
}

function isPrime(n) {
  let isPrime = true;

  if (n <= 1) return false;

  for (let i = 2; i < n; i++) {
    if (n % i == 0) {
      isPrime = false;
      break;
    }
  }
  return isPrime
    ? console.log("It is a prime number")
    : console.log("It is not a prime number");
}

factorial(5);
isPrime(12);
fibonacciSeries(10);

File: pratice/LogicalCoding/MathProblems.py
from sympy import *


def fibonacci(n):
    prev1 = current = 0
    prev2 = 1
    sequence = [prev1, prev2]

    for _ in range(2, n):
        current = prev1 + prev2
        sequence.append(current)
        prev1 = prev2
        prev2 = current
    print(sequence)


def factorial(n):
    if n == 0 or n == 1:
        return 1
    else:
        value = 1
        for i in range(2, n + 1):
            value *= i
        return value
        # return n * factorial(n-1) // same as above but with recursion


def is_prime(n):
    isPrime = True

    if n <= 1:
        return False
    for i in range(2, n):
        if n % i == 0:
            isPrime = False
            break
    # return isPrime
    return isprime(n)  # using sympy library


fibonacci(10)
print(factorial(5))
print(is_prime(12))
